Gemfile

```diff
@@ -51,11 +51,11 @@ gem '3scale_time_range', '0.0.6'
 gem 'statsd-ruby', require: false
 
 # Sidekiq
-gem 'sidekiq', '~> 6.4.0', require: %w[sidekiq sidekiq/web]
-gem 'sidekiq-batch'
+gem 'sidekiq', '~> 7', require: %w[sidekiq sidekiq/web]
+gem 'sidekiq-batch', github: '3scale/sidekiq-batch', branch: 'redis-client'
```

Comment:
`sidekiq-batch` is not compatible with `redis-client`, which is required by Sidekiq 7. 

The gem seems abandoned and it's not like it's ever going to add support for `redis-client`. Fortunately, the gem is small and only needs a few tweaks to support it. 

Instead of creating a monkey patch which could become too big, I created a fork to add support for Sidekiq 7/redis-client. See the [diff](https://github.com/breamware/sidekiq-batch/compare/master...3scale:sidekiq-batch:redis-client)

I made the tests pass so I'm confident enough.

Reply:
shouldn't redis deps be taken care of by the sidekiq deps, not specified there:
https://github.com/breamware/sidekiq-batch/compare/master...3scale:sidekiq-batch:redis-client#diff-6a4117dccc1ba4a2356da2527865a7da697e5df4680090ef8641f6a14e8df95cR23

Reply:
I guess you wonder why is `redis` a dependency of `sidekiq-batch` in my fork but not originally. I'll try to explain:

1. Sidekiq 7 depends on `redis-client`
2. Sidekiq-batch depends on `fakeredis`
3. `fakeredis` doesn't support `redis-client`, so `sidekiq-batch` tests fail for sidekiq 7
4. To replace `fakeredis`, I need to launch a redis instance locally and use the `redis` gem, so the `sidekiq-batch` tests can run.
5. The `redis` gem depends on `redis-client`, so `sidekiq-batch` tests use it.
6. The `hiredis-client` gem is the redis driver we use on porta, so I though tests would be more relevant for us if I reproduce the porta gemset for `sidekiq-batch` tests.

Since it's only for tests. I can probably move `redis` and `hiredis-client` to the development dependencies section. I'll try.

I hope what I wrote makes sense. It's not easy, considering all Redis related gems out there have basically the same name 😵‍💫.

Reply:
Yeah, not very clear but moving these to dev/test deps + a comment would make it clearer. Also will not pull these into porta.

Reply:
So although it is not about a change in this PR but in another repo, the idea is what we could update these two to development dependencies, correct?

https://github.com/3scale/sidekiq-batch/blob/da79c4332f9db464db9c09a7bca30f4a2631520c/sidekiq-batch.gemspec

Reply:
I'm pushing about this because changing that should get the gems out of our `Gemfile.lock`

Reply:
yes, I'll move them to development dependencies. Anyway, they won't be out of our Gemfile.lock because porta uses redis itself, not only for sidekiq, so we need them anyway.

Reply:
Done